magicctapipe/io/io.py

```diff
@@ -172,6 +219,91 @@ def get_stereo_events(
     return event_data_stereo
 
 
+
+def get_stereo_events(
+    event_data, config, quality_cuts=None, group_index=["obs_id", "event_id"], eval_multi_combo=True
+):
+    """
+    Gets the stereo events surviving specified quality cuts.
+
+    It also adds the telescope multiplicity `multiplicity` and
+    combination types `combo_type` to the output data frame.
+
+    Parameters
+    ----------
+    event_data: pandas.core.frame.DataFrame
+        Data frame of shower events
+    config: dict 
+        Read from the yaml file with information about the telescope IDs. Typically called "config_general.yaml"
+    quality_cuts: str
+        Quality cuts applied to the input data
+    group_index: list
+        Index to group telescope events
+    
+
+    Returns
+    -------
+    event_data_stereo: pandas.core.frame.DataFrame
+        Data frame of the stereo events surviving the quality cuts
+    """
+    
+    TEL_NAMES, TEL_COMBINATIONS = telescope_combinations(config)
+    
+    event_data_stereo = event_data.copy()
+
+    # Apply the quality cuts
+    if quality_cuts is not None:
+        event_data_stereo.query(quality_cuts, inplace=True)
+    
+    max_multiplicity=len(TEL_NAMES.keys())
+    # Extract stereo events
+    event_data_stereo["multiplicity"] = event_data_stereo.groupby(group_index).size()
+    event_data_stereo.query(f"multiplicity >1 & multiplicity <= {max_multiplicity}", inplace=True)
+    if eval_multi_combo==True:
+        # Check the total number of events
+        n_events_total = len(event_data_stereo.groupby(group_index).size())
+        logger.info(f"\nIn total {n_events_total} stereo events are found:")
+
+        n_events_per_combo = {}
+
+        # Loop over every telescope combination type
+        for combo_type, (tel_combo, tel_ids) in enumerate(TEL_COMBINATIONS.items()):
```

Comment:
I think it would be good to extract this part  recalculating multiplicity into a separate function. You are already using it twice in this file, and we need to do it basically everywhere where cuts on individual telescopes are done. it can take as argument df_events and TEL_COMBINATIONS and return df_events and this dictionary with statistics of combo types

Reply:
We use it twice because we are keeping the function get_stereo_events_old() in this PR. This function is necessary only for the combo-type RF and will definitely disappear in the next PR, where we have a telescope-wise RF. Such that in the end, we will use it only once. We also don't use this loop in the other scripts, so we think we can keep it as it is at least for now.

Reply:
I actually had a situation in performance scripts when I had to recalculate it and a separate function would be useful, but this is not a big issue, we might always add it in some future PR if we want.